apps-api/backoffice/src/modules/topic/index.ts

```diff
@@ -0,0 +1,157 @@
+import Elysia from 'elysia'
+
+import {
+  FollowTopicParams,
+  FollowTopicResponse,
+  GetTopicsResponse,
+  UnFollowTopicParams,
+} from './models'
+import { TopicServicePlugin } from './service'
+
+import { InternalErrorCode } from '../../dtos/error'
+import { AuthGuardPlugin } from '../../plugins/auth-guard'
+import { createErrorSchema, exhaustiveGuard, mapErrorCodeToResponse } from '../../utils/error'
+import { UnfollowUserResponse } from '../profile/models'
+
+export const TopicController = new Elysia({
+  prefix: '/topics',
+  tags: ['Topics'],
+})
+  .use(TopicServicePlugin)
+  .use(AuthGuardPlugin)
+  .get(
+    '/',
+    async ({ topicService, status }) => {
+      const topics = await topicService.getTopics()
+
+      if (topics.isErr()) {
+        return mapErrorCodeToResponse(topics.error, status)
+      }
+
+      return status(200, topics.value)
+    },
+    {
+      response: {
+        200: GetTopicsResponse,
+        ...createErrorSchema(InternalErrorCode.INTERNAL_SERVER_ERROR),
+      },
+      detail: {
+        summary: 'Get all topics',
+        description: 'Get all topics entries',
+      },
+    }
+  )
+  .get(
+    '/follows',
+    async ({ user, status, topicService }) => {
+      const topics = await topicService.getFollowedTopics(user.id)
+
+      if (topics.isErr()) {
+        return mapErrorCodeToResponse(topics.error, status)
+      }
+
+      return status(200, topics.value)
+    },
+    {
+      requiredLocalUser: true,
+      response: {
+        200: GetTopicsResponse,
+        ...createErrorSchema(
+          InternalErrorCode.INTERNAL_SERVER_ERROR,
+          InternalErrorCode.UNAUTHORIZED
+        ),
+      },
+      detail: {
+        summary: 'Get followed topics',
+        description: 'Get followed topics',
+      },
+    }
+  )
+  .post(
+    '/:topicId/follow',
+    async ({ params, user, topicService, status }) => {
+      const topicId = params.topicId
+      const userId = user.id
+      const result = await topicService.followTopic(topicId, userId)
+
+      if (result.isErr()) {
+        switch (result.error.code) {
+          case InternalErrorCode.TOPIC_NOT_FOUND:
+            return mapErrorCodeToResponse(result.error, status)
+          case InternalErrorCode.TOPIC_CANNOT_FOLLOW_DRAFT:
+            return mapErrorCodeToResponse(result.error, status)
+          case InternalErrorCode.TOPIC_ALREADY_FOLLOWED:
+            return mapErrorCodeToResponse(result.error, status)
+          case InternalErrorCode.INTERNAL_SERVER_ERROR:
+            return mapErrorCodeToResponse(result.error, status)
+          default:
+            exhaustiveGuard(result.error)
```

Comment:
Please update `main` branch and change logic to

```suggestion
		return mapErrorCodeToResponse(result.error, status)
```